server/api/controllers/instances/instancesController.js

```diff
@@ -42,13 +49,54 @@ function getInstances(req, res, next) {
     if (instanceId !== undefined) {
       filter['instance-id'] = instanceId;
     }
-
+    // TODO(Filip): consider adding filter on launch-time for improved performance
+    // if (since !== undefined) {
+    //   filter['launch-time'] = Instance.createLaunchTimeArraySince(since);
+    // }
+    
     if (_.isEmpty(filter)) {
       filter = null;
     }
 
     let handler = accountName !== undefined ? ScanInstances : ScanCrossAccountInstances;
-    handler({ accountName, filter }).then((data) => res.json(data))
+    let list = yield handler({ accountName, filter });
+
+    // Note: be wary of performance - this filters instances AFTER fetching all from AWS
+    if (since !== undefined) {
+      let sinceDate = new Date(since);
+      list = _.filter(list, (instance) => {
+        return sinceDate.getTime() < new Date(instance.LaunchTime).getTime();
+      })
+    }
+
+    if (includeServices === true) {
+      list = yield _.map(list, (instance) => {
+        let instanceEnvironment = instance.getTag('Environment', null);
+        
+        instance.appendTagsToObject();
+
+        let instanceName = instance.getTag('Name', null);
+        if (instanceName === null || instanceEnvironment === null) {
+          // This instance won't be returned
+          return false;
+        }
+
+        return Environment.getAccountNameForEnvironment(instanceEnvironment).then((accountName) => {
```

Comment:
Environment.getAccountNameForEnvironment is not currently cached and here called in tight loop over all instances.